File: jars/HerbPack/src/main/java/herb/herb/commands/manhuntCommand.java
package herb.herb.commands;

import herb.herb.utilitys.GlobalArgus;
import net.kyori.adventure.text.Component;
import net.kyori.adventure.title.TitlePart;
import org.bukkit.Bukkit;
import org.bukkit.ChatColor;
import org.bukkit.GameMode;
import org.bukkit.command.Command;
import org.bukkit.command.CommandSender;
import org.bukkit.command.TabExecutor;
import org.bukkit.entity.Player;
import org.bukkit.scheduler.BukkitScheduler;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import java.util.ArrayList;
import java.util.List;

public class manhuntCommand implements TabExecutor {
    @Override
    public boolean onCommand(@NotNull CommandSender sender, @NotNull Command command, @NotNull String label, @NotNull String[] args) {
        GlobalArgus Gargs = new GlobalArgus();
        @NotNull List<Player> p = new ArrayList<Player>();
        try {
            p.add(Bukkit.getPlayer(args[0]));
        }catch (Exception e){
            GlobalArgus.ServerConfig.getComments("teams."+args[0]).forEach(pl -> p.add(Bukkit.getPlayer(pl)));
        }
        try {
            Bukkit.getOnlinePlayers().forEach(pl -> {
                pl.teleport(p.get(0).getLocation());
                pl.setGameMode(GameMode.CREATIVE);
            });
            BukkitScheduler sc = Bukkit.getPluginManager().getPlugin("HerbPack").getServer().getScheduler();
            sc.scheduleSyncDelayedTask(Bukkit.getPluginManager().getPlugin("HerbPack"), new Runnable() {
                        @Override
                        public void run() {
                            GlobalArgus.wait(3,1);
                            GlobalArgus.wait(2,2);
                            GlobalArgus.wait(1,3);
                            Bukkit.getOnlinePlayers().forEach(pl -> {pl.sendTitlePart(TitlePart.SUBTITLE, Component.text(ChatColor.YELLOW+"시작"));});
                        }
                    },20L);

        } catch (Exception e) {}
        Gargs.ManHunttarget = p;
        Gargs.Manhuntgamestart = true;
        Bukkit.getOnlinePlayers().forEach(pl -> {
            pl.setGameMode(GameMode.SURVIVAL);
            Gargs.Manhuntbar.keySet().forEach(player -> Gargs.Manhuntbar.get(player).addPlayer(pl));
        });
        Gargs.PlayStartTime = System.currentTimeMillis();

        return true;
    }

    @Override
    public @Nullable List<String> onTabComplete(@NotNull CommandSender sender, @NotNull Command command, @NotNull String alias, @NotNull String[] args) {
        if (command.getName().equals("startmanhunt")){
            if (args.length == 1){
                List playerList = new ArrayList<>();
                Bukkit.getOnlinePlayers().forEach(p -> playerList.add(p.getName()));
                return playerList;
            }else if (args.length == 2){
                List typeList = new ArrayList<>();
                GlobalArgus.ServerConfig.getKeys(true);
                typeList.add("nomal");
                typeList.add("randomEffect");
                return typeList;
            }
        }
        return null;
    }
}

File: jars/HerbPack/src/main/java/herb/herb/commands/manhuntCommand.kt
package herb.herb.commands

import herb.herb.Argus.GlobalArgus.Companion.wait
import org.bukkit.command.TabExecutor
import herb.herb.Argus.GlobalArgus
import herb.herb.Events.ManhuntEvent
import org.bukkit.entity.Player
import java.lang.Runnable
import net.kyori.adventure.title.TitlePart
import net.kyori.adventure.text.Component
import org.bukkit.*
import org.bukkit.command.Command
import org.bukkit.command.CommandSender
import java.lang.Exception
import java.util.function.Consumer
import kotlin.collections.ArrayList

class manhuntCommand() : TabExecutor {
    override fun onCommand(sender: CommandSender, command: Command, label: String, args: Array<String>): Boolean {
        GlobalArgus.pm.registerEvents(ManhuntEvent(), (GlobalArgus.plugin)!!)
        var p:Player? = Bukkit.getPlayer(args[0])
        if (p == null){
            sender.sendMessage("존재하지 않는 플레이어입니다")
        }else{
            Bukkit.getOnlinePlayers().forEach { pl: Player ->
                pl.teleport(p.getLocation())
                pl.setGameMode(GameMode.CREATIVE)
            }
            val sc = Bukkit.getPluginManager().getPlugin("HerbPack")!!.server.scheduler
            sc.scheduleSyncDelayedTask((Bukkit.getPluginManager().getPlugin("HerbPack"))!!, Runnable {
                wait(3, 1)
                wait(2, 2)
                wait(1, 3)
                Bukkit.getOnlinePlayers().forEach { pl: Player -> pl.sendTitlePart(TitlePart.SUBTITLE, Component.text(ChatColor.YELLOW.toString() + "시작")) } }, 20L)
        }
        GlobalArgus.ManHunttarget = p
        GlobalArgus.Manhuntgamestart = true
        Bukkit.getOnlinePlayers().forEach { pl: Player ->
            pl.setGameMode(GameMode.SURVIVAL)
            GlobalArgus.Manhuntbar.addPlayer(pl)
        }
        GlobalArgus.PlayStartTime = System.currentTimeMillis()
        return true
    }

    override fun onTabComplete(sender: CommandSender, command: Command, alias: String, args: Array<String>): List<String>? {
        if ((command.name == "startmanhunt")) {
            if (args.size == 1) {
                val playerList: MutableList<String> = ArrayList<String>()
                Bukkit.getOnlinePlayers().forEach { p: Player -> playerList.add(p.getName()) }
                return playerList
            } else if (args.size == 2) {
                val typeList: MutableList<String> = ArrayList<String>()
                GlobalArgus.ServerConfig!!.getKeys(true)
                typeList.add("nomal")
                typeList.add("randomEffect")
                return typeList
            }
        }
        return null
    }
}